src/mocks/msw/handlers.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  BackendDestinationEnum,
+  getEndpoint,
+} from '../../lib/utils/functions/getEndpoint'
+import { mockSchemaBinding } from '../mockSchema'
+import { rest } from 'msw'
+import {
+  mockFileEntityBundle,
+  mockProjectEntityBundle,
+  MOCK_FILE_ENTITY_ID,
+  MOCK_PROJECT_ID,
+} from '../entity/mockEntity'
+import {
+  mockUserBundle,
+  mockUserProfileData,
+  MOCK_USER_ID,
+} from '../user/mock_user_profile'
+import {
+  ENTITY_BUNDLE_V2,
+  ENTITY_SCHEMA_BINDING,
+  USER_ID_BUNDLE,
+  USER_PROFILE_ID,
+} from '../../lib/utils/APIConstants'
+
+const handlers = [
+  rest.options('*', async (req, res, ctx) => {
+    return res(ctx.status(200))
+  }),
+
+  rest.get(
+    `${getEndpoint(BackendDestinationEnum.REPO_ENDPOINT)}${USER_PROFILE_ID(
+      ':id',
+    )}`,
+    async (req, res, ctx) => {
+      let response = {}
+      let status = 404
+      if (req.params.id === MOCK_USER_ID.toString()) {
+        response = mockUserProfileData
+        status = 200
+      }
+      return res(ctx.status(status), ctx.json(response))
+    },
+  ),
+
+  rest.get(
+    `${getEndpoint(BackendDestinationEnum.REPO_ENDPOINT)}${USER_ID_BUNDLE(
+      ':id',
+    )}`,
+    async (req, res, ctx) => {
+      let response = {}
+      let status = 404
+      if (req.params.id === MOCK_USER_ID.toString()) {
+        response = mockUserBundle
+        status = 200
+      }
+      return res(ctx.status(status), ctx.json(response))
+    },
+  ),
+
+  rest.post(
+    `${getEndpoint(BackendDestinationEnum.REPO_ENDPOINT)}${ENTITY_BUNDLE_V2(
+      ':entityId',
+    )}`,
+    async (req, res, ctx) => {
+      let response = {}
+      let status = 404
+      if (req.params.entityId === MOCK_FILE_ENTITY_ID) {
+        response = mockFileEntityBundle
+        status = 200
+      } else if (req.params.entityId === MOCK_PROJECT_ID) {
+        response = mockProjectEntityBundle
+        status = 200
+      }
+      return res(ctx.status(status), ctx.json(response))
+    },
+  ),
```

Comment:
Here, we map specific IDs to specific mocks. The ID constants (`MOCK_FILE_ENTITY_ID`) are exposed so tests can use them.

This isn't used in tests in this PR, but I'm wondering if this seems like a good pattern to use, or if it will become a headache to maintain our mocks.